operate/account/user.py

```diff
@@ -49,24 +50,44 @@ def load(cls, path: Path) -> "UserAccount":
     def new(cls, password: str, path: Path) -> "UserAccount":
         """Create a new user."""
         user = UserAccount(
-            password_sha=sha256(string=password),
+            password_hash=argon2id(password=password),
             path=path,
         )
         user.store()
         return UserAccount.load(path=path)
 
     def is_valid(self, password: str) -> bool:
         """Check if a password string is valid."""
-        return sha256(string=password) == self.password_sha
+        try:
+            ph = argon2.PasswordHasher()
+            valid = ph.verify(self.password_hash, password)
+
+            if valid and ph.check_needs_rehash(self.password_hash):
+                self.password_hash = argon2id(password)
+                self.store()
+
+            return valid
+        except argon2.exceptions.VerificationError:
+            return False
+        except argon2.exceptions.InvalidHashError:
+            # Verify legacy password hash and update it to Argon2id if valid
+            sha256 = hashlib.sha256()
+            sha256.update(password.encode())
+            if sha256.hexdigest() == self.password_hash:
+                self.password_hash = argon2id(password=password)
+                self.store()
+                return True
```

Comment:
this logic is a migration from the legacy password, so wdyt about keeping it in the MigrationManager?

Reply:
The idea of the migration manager is to "rearrange" the contents of the .json files with the information contained therein (or "public" information) so that it matches the expected logic by the classes. This way, we don't have to 'pollute' code on classes like service.py, etc with migrating formats, and thus it will be easier to reason about the code).

Updating the hash is somehow an exception, because this can only be updated once we know the password (and this only happens when the user logs in).